vowpalwabbit/core/src/reductions/cs_active.cc

```diff
@@ -302,12 +304,77 @@ void predict_or_learn(cs_active& cs_a, single_learner& base, VW::example& ec)
   ec.l.cs = ld;
 }
 
-void finish_example(VW::workspace& all, cs_active&, VW::example& ec)
+void update_stats_cs_active(const VW::workspace& /* all */, VW::shared_data& sd, const cs_active& /* data */,
+    const VW::example& ec, VW::io::logger& logger)
 {
-  VW::details::output_cs_example(all, ec, ec.l.cs, ec.pred.active_multiclass.predicted_class);
-  VW::finish_example(all, ec);
+  const auto& label = ec.l.cs;
+  const auto multiclass_prediction = ec.pred.active_multiclass.predicted_class;
```

Comment:
should this be a ref?

Reply:
nm, looks like an int type from code below... one of the few things i hate about auto